config/initializers/semantic_logger.rb

```diff
@@ -0,0 +1,55 @@
+class LogStashFormatter < SemanticLogger::Formatters::Raw
+  def call(log, logger)
+    super(log, logger)
+    format_add_type
+    format_job_data
+    format_duration
+    format_exception
+    format_json_message_context
+    format_backtrace
+    hash.to_json
+  end
+
+private
+
+  def format_add_type
+    hash[:type] = "rails"
+  end
+
+  def format_job_data
+    hash[:job_id] = RequestStore.store[:job_id] if RequestStore.store[:job_id].present?
+    hash[:job_queue] = RequestStore.store[:job_queue] if RequestStore.store[:job_queue].present?
+  end
+
+  def format_duration
+    hash[:duration] = hash[:duration_ms]
+    hash[:duration_ms] = nil
+  end
+
+  def format_exception
+    exception_message = hash.dig(:exception, :message)
+    if exception_message.present?
+      hash[:message] = "Exception occured: #{exception_message}"
+    end
+  end
+
+  def format_json_message_context
+    if hash[:message].present?
+      context = JSON.parse(hash[:message])["context"]
+      hash[:sidekiq_job_context] = hash[:message]
+      hash[:message] = context
+    end
+  rescue JSON::ParserError
+    nil
+  end
+
+  def format_backtrace
+    if hash[:message]&.start_with?("/")
+      message_lines = hash[:message].split("\n")
+      if message_lines.all? { |line| line.start_with?("/") }
+        hash[:backtrace] = hash[:message]
+        hash[:message] = "Exception occured: #{message_lines.first}"
+      end
+    end
+  end
+end
```

Comment:
Do we want to enable this already for the production environment?

Reply:
Yeah, it might be worth checking as a smoke test to see if it's logging the way we expect

Reply:
I think my point is that we need this section:

https://github.com/DFE-Digital/teacher-training-api/blob/master/config/initializers/semantic_logger.rb#L51-L66

Reply:
Done in https://github.com/DFE-Digital/register-trainee-teacher-data/pull/3/commits/7552f05a8a21009c15fcfd6ac6b24aa00cb8b440